src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/FixLegacyResourceDesignerStep.cs

```diff
@@ -90,6 +77,15 @@ internal override bool ProcessAssemblyDesigner (AssemblyDefinition assembly)
 				return false;
 			}
 
+			// This is expected for the first call, in <LinkAssembliesNoShrink/>
+			if (!designerLoaded)
+				LoadDesigner ();
+
+			if (designerAssembly == null || designerType == null) {
+				LogMessage ($"   Not using {DesignerAssemblyName}");
+				return false;
+			}
+
```

Comment:
This is the main change that helps perf, delay the `LoadDesigner()` call to here. I also had to move the `designerAssembly ` and `designerType ` null checks down for it to actually work...